src/game/stage/camera.c

```diff
@@ -0,0 +1,645 @@
+    bgs->unkC0.unk28 = unkA98[1];
+
+    gUnknown_03004D80[1] = 0;
+    gUnknown_03002280[1][0] = 0;
+    gUnknown_03002280[1][1] = 0;
+    gUnknown_03002280[1][2] = 0xff;
+    gUnknown_03002280[1][3] = 0x20;
+    gUnknown_03004D80[2] = 0;
+    gUnknown_03002280[2][0] = 0;
+    gUnknown_03002280[2][1] = 0;
+    gUnknown_03002280[2][2] = 0xff;
+    gUnknown_03002280[2][3] = 0x20;
+
+    if (gGameMode == GAME_MODE_MULTI_PLAYER_COLLECT_RINGS) {
+        bgs->unk40.unk2E |= 0x208;
+        bgs->unk80.unk2E |= 0x208;
+    }
+
+    if (level != 0x1D) {
+        sub_8002A3C(&bgs->unk40);
+        sub_8002A3C(&bgs->unk80);
+        sub_8002A3C(&bgs->unkC0);
+    }
+
+    if (gGameMode != GAME_MODE_MULTI_PLAYER_COLLECT_RINGS) {
+        gUnknown_030059C8 = gCollisionTable[level];
+    } else {
+        gUnknown_030059C8 = *(u32 **)(EWRAM_START + 0x33004);
+    }
+
+    camera->unk28 = 0;
+    camera->unk2C = gUnknown_030059C8[8];
+    camera->unk30 = 0;
+    camera->unk34 = gUnknown_030059C8[7];
+
+    if (((gCurrentLevel & ACTS_PER_ZONE) == ACT_BOSS)
+        || ((gCurrentLevel == 0x1C) && (gUnknown_030054B0 == 0))
+        || (gCurrentLevel == 0x1D)) {
+        if (gCurrentLevel == 0x1D) {
+            sub_802C668(&player->x, &player->y);
+            gUnknown_03005440 = gUnknown_080D5964[LEVEL_TO_ZONE(0x20)][0];
+            gUnknown_030054BC = gUnknown_080D5964[LEVEL_TO_ZONE(0x20)][1];
+            camera->x = 600;
+            camera->unk10 = 0x78;
+            camera->unk14 = 0;
+            camera->y = 0;
+            camera->unk64 = 0xFFFC;
+        } else {
+            camera->x = Q_24_8_TO_INT(player->x);
+            camera->unk10 = Q_24_8_TO_INT(player->x) - 0x1E0;
+            camera->y = Q_24_8_TO_INT(player->y) - 0x54;
+            camera->unk14 = camera->y;
+            camera->unk64 = player->unk17 - 4;
+        }
+    } else {
+        camera->x = Q_24_8_TO_INT(player->x) - 0x78;
+        camera->y = Q_24_8_TO_INT(player->y) - 0x54;
+
+        if (camera->x < 0) {
+            camera->x = 0;
+        }
+
+        if (camera->y < 0) {
+            camera->y = 0;
+        }
+
+        camera->unk10 = camera->x;
+        camera->unk14 = camera->y;
+        camera->unk64 = player->unk17 - 4;
+    }
+
+    camera->unk8 = 0x1000;
+    camera->unkC = 0xc;
+    camera->unk56 = 0;
+    camera->shiftX = 0;
+    camera->shiftY = 0;
+    camera->unk20 = 0;
+    camera->unk24 = 0;
+    camera->unk40 = 0;
+    camera->unk44 = 8;
+    camera->unk48 = 0;
+    camera->unk4C = 0;
+    camera->unk50 = 0;
+    camera->unk52 = 0;
+    camera->unk54 = 0;
+    camera->unk60 = 0;
+    camera->unk62 = 0;
+
+    camera->movementTask = TaskCreate(Task_801E0A8, 0, 0xF00, 0, TaskDestructor_801E040);
+
+    camera->unk58 = sStageBgUpdateFuncs[level];
+
+    if (sStageInitProcedures[level] != NULL) {
+        sStageInitProcedures[level]();
+    }
+}
+
+void sub_801C708(s32, s32);
+
+#define BOSS_CAM_FRAME_DELTA_PIXELS 5
+
+void sub_801C394(void)
```

Comment:
CameraMain?

Reply:
It's not a task, but UpdateCamera should suffice, I think?